components/data-feed-service/service/data-feed-aggregate-task.go

```diff
@@ -0,0 +1,179 @@
+package service
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	cfgmgmt "github.com/chef/automate/api/interservice/cfgmgmt/service"
+	"github.com/chef/automate/components/compliance-service/api/reporting"
+	"github.com/chef/automate/components/data-feed-service/config"
+	"github.com/chef/automate/lib/cereal"
+	"github.com/chef/automate/lib/grpc/secureconn"
+)
+
+var (
+	dataFeedAggregateTaskName = cereal.NewTaskName("data-feed-aggregate")
+)
+
+type DataFeedAggregateTask struct {
+	cfgMgmt   cfgmgmt.CfgMgmtClient
+	reporting reporting.ReportingServiceClient
+}
+
+type DataFeedAggregateTaskResults struct {
+	DataFeedMessages map[string]map[string]interface{}
+	NodeIDs          map[string]NodeIDs
+}
+
+func NewDataFeedAggregateTask(dataFeedConfig *config.DataFeedConfig, connFactory *secureconn.Factory) (*DataFeedAggregateTask, error) {
+
+	cfgMgmtConn, err := connFactory.Dial("config-mgmt-service", dataFeedConfig.CfgmgmtConfig.Target)
```

Comment:
I think it might be better if these constructors took the relevant connections as arguments, that way we don't need to create & manage multiple connection objects for the same upstream.

Reply:
fair enough